File: src/tools/wss.ts
import { z } from "zod";
import { runBridge } from "../bridge/runner.js";

export const wssTool = {
  name: "wind_wss",
  description: "Fetch snapshot / point-in-time values for securities (e.g. latest price, PE ratio) using Wind w.wss().",
  inputSchema: z.object({
    codes: z.string().describe("Comma-separated Wind security codes"),
    fields: z.string().describe("Comma-separated data fields"),
    options: z.string().optional().describe("Wind options string"),
  }),
  handler: async (params: Record<string, unknown>) => {
    const result = await runBridge({ function: "wss", params });
    if (!result.ok) throw new Error(result.error);
    return result.data;
  },
};

File: src/python/handlers/wss.py
from utils import wind_data_to_dict, handle_error
from WindPy import w


def handle_wss(params: dict) -> dict:
    codes = params["codes"]
    fields = params["fields"]
    options = params.get("options", "")

    data = w.wss(codes, fields, options)

    if data.ErrorCode != 0:
        raise RuntimeError(handle_error(data.ErrorCode))

    return wind_data_to_dict(data)
